static/js/panels/ProfileForm.tsx

```diff
@@ -28,7 +52,71 @@ const ProfileForm: FC = () => {
     },
     validationSchema: ProfileSchema,
     onSubmit: (values) => {
-      createProfile(values.name, values.description)
+      // create LxdDevices object
+      const devicesObj: LxdDevices = {};
+      if (nicDevice.network) {
+        devicesObj[nicDevice.name === "" ? nicDevice.network : nicDevice.name] =
+          nicDevice;
+      }
+      if (rootDiskDevice.pool) {
+        devicesObj.root = rootDiskDevice;
+      }
+
+      // create LxdConfigPair object for limits
+      const limits: LxdConfigPair = {};
+      switch (cpuLimit.selectedType) {
```

Comment:
I'd push this complexity into a function to  make it more readable:

```
Index: static/js/panels/ProfileForm.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/static/js/panels/ProfileForm.tsx b/static/js/panels/ProfileForm.tsx
--- a/static/js/panels/ProfileForm.tsx	(revision 706eb18f5a4b51c0b994db09c7217d8886974d55)
+++ b/static/js/panels/ProfileForm.tsx	(date 1671111390214)
@@ -27,6 +27,27 @@
   DEFAULT_NIC_DEVICE,
 } from "../util/defaults";
 
+const getCpuLimit = (cpuLimit: CpuLimit) => {
+  switch (cpuLimit.selectedType) {
+    case CPU_LIMIT_TYPE.DYNAMIC:
+      if (cpuLimit.dynamicValue) {
+        return `${cpuLimit.dynamicValue}`;
+      }
+      return null;
+    case CPU_LIMIT_TYPE.FIXED_RANGE:
+      if (cpuLimit.rangeValue?.from && cpuLimit.rangeValue.to) {
+        return `${cpuLimit.rangeValue.from}-${cpuLimit.rangeValue.to}`;
+      }
+      return null;
+    case CPU_LIMIT_TYPE.FIXED_SET:
+      if (cpuLimit.setValue) {
+        return cpuLimit.setValue;
+      }
+      return null;
+  }
+  return null;
+};
+
 const ProfileForm: FC = () => {
   const navigate = useNavigate();
   const notify = useNotification();
@@ -64,31 +85,11 @@
 
       // create LxdConfigPair object for limits
       const limits: LxdConfigPair = {};
-      switch (cpuLimit.selectedType) {
-        case CPU_LIMIT_TYPE.DYNAMIC:
-          if (cpuLimit.dynamicValue) {
-            limits["limits.cpu"] = `${cpuLimit.dynamicValue}`;
-          }
-          break;
-        case CPU_LIMIT_TYPE.FIXED_RANGE:
-          if (!cpuLimit.rangeValue) break;
-          if (
-            cpuLimit.rangeValue.from !== null &&
-            cpuLimit.rangeValue.to !== null
-          ) {
-            limits[
-              "limits.cpu"
-            ] = `${cpuLimit.rangeValue.from}-${cpuLimit.rangeValue.to}`;
-          }
-          break;
-        case CPU_LIMIT_TYPE.FIXED_SET:
-          if (cpuLimit.setValue) {
-            limits["limits.cpu"] = cpuLimit.setValue;
-          }
-          break;
-        default:
-          break;
+      const limitsCpu = getCpuLimit(cpuLimit);
+      if (limitsCpu) {
+        limits["limits.cpu"] = limitsCpu;
       }
+
       if (memoryLimit.value) {
         limits["limits.memory"] = `${memoryLimit.value}${memoryLimit.unit}`;
       }
```

Reply:
Little bug in there (also in the patch I proposed): The setValue can be a single number and needs to be turned into something like 1-1 in that case.

Reply:
Good idea to move the `switch` to a function, but your patch actually introduces a bug: when the value in the `from`/`to` field is set to 0 (which is a possible and allowed value for these fields), the expression `if (cpuLimit.rangeValue?.from && cpuLimit.rangeValue.to)` will equal to `false` and will return `null`, which is wrong.
In other words: I wasn't putting the explicit `null` checks without a reason. ;)

Reply:
Right.